common/entryBlock/entry.go

```diff
@@ -143,7 +143,7 @@ func (c *Entry) DatabasePrimaryIndex() (rval interfaces.IHash) {
 
 // DatabaseSecondaryIndex always returns nil (ie, no secondary index)
 func (c *Entry) DatabaseSecondaryIndex() (rval interfaces.IHash) {
-	defer func() { rval = primitives.CheckNil(rval, "Entry.DatabaseSecondaryIndex") }()
+	defer func() { rval = primitives.CheckNilInterface(rval, "Entry.DatabaseSecondaryIndex") }()
 
 	return nil
 }
```

Comment:
I don't understand what this pattern does. What's the purpose of this defer function?

Reply:
Nothing in this instance, I just did a find/replace without looking at the placement in-depth.

Reply:
Ah, you're right, it seems rather useless.